src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -2731,9 +2737,21 @@
     <TypeComment>
       <summary>Member declaration syntax.</summary>
     </TypeComment>
+    <Field Name="AttributeLists" Type="SyntaxList&lt;AttributeListSyntax&gt;">
+      <PropertyComment>
+        <summary>Gets the attribute declaration list.</summary>
+      </PropertyComment>
+    </Field>
+    <Field Name="Modifiers" Type="SyntaxList&lt;SyntaxToken&gt;">
+      <PropertyComment>
+        <summary>Gets the modifier list.</summary>
+      </PropertyComment>
+    </Field>
   </AbstractNode>
   <Node Name="NamespaceDeclarationSyntax" Base="MemberDeclarationSyntax">
     <Kind Name="NamespaceDeclaration"/>
+    <Field Name="AttributeLists" Type="SyntaxList&lt;AttributeListSyntax&gt;" Override="true"/>
```

Comment:
><Field Name="AttributeLists" Type="SyntaxList&lt;AttributeListSyntax&gt;" Override="true"/> [](start = 4, length = 91)

I'm not sure if that's needed or useful. Let's figure out what would happen if `AttributeLists`/`Modifiers` were only defined on `MemberDeclarationSyntax`.

Reply:
it's actually helpful since hte parser *already* has to deal with this today, and it does it by jamming on skipped tokens onto nodes.  It's much cleaner to just have a tree where the nodes are properly attached, even if it represents a superset of C#. 

Note: this is what TS does as well and it makes things nice and clean too.

Reply:
Sorry I wasn't clear. What I meant is that `MemberDeclarationSyntax` has both fields already. What happens if we delete these two lines from `NamespaceDeclarationSyntax` and other children of `MemberDeclarationSyntax`?

---
In reply to: [258161112](https://github.com/dotnet/roslyn/pull/32999#discussion_r258161112) [](ancestors = 258161112)

Reply:
Ah.  I believe that doesn't work.  If you get rid of them, tehn you don't get the overrides and the code doesn't compile.  That's how i started, but i had to add all the fields to all the concrete subclasses.  I think this is just how syntax.xml works!